### PR TITLE
special-pages: integrated watch mode

### PR DESCRIPTION
**Pull request by shakyShane**
**Asana Task/Github Issue:** 

## Description

- [x] added optional watch mode for special pages
  - [x] CSS changes will be injected
  - [x] rebuilds are instant and localised to the page in question
- [x] use ‘public’ + ’src’ to make it clear which files will be copied to the build output

** Before **

When running `npm run watch` - changes to ANY file would cause every special-page to be rebuilt from scratch. 

This worked ok with few pages and very little assets. But now as the projects grow in size and complexity, we need to consider them as isolated applications in terms of development.

** After **

When running `npm run watch — —page=onboarding`, only files in the onboarding project are watched 💪🏻.

## Testing Steps

- [ ] locally, inside `special-pages`, run `npm run watch — —page=onboarding` (or any other page)
- [ ] open localhost:8000 to view the page in question

## Checklist

*Please tick all that apply:*

- [ ] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#1317** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/1317?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: special-pages/pages/example/src/index.js [hunk 1]
```diff
@@ -5,9 +5,9 @@
  */
 
 import { createTypedMessages } from '@duckduckgo/messaging';
-import { Environment } from '../../../../shared/environment.js';
-import { createSpecialPageMessaging } from '../../../../shared/create-special-page-messaging.js';
-import { init } from '../../app/index.js';
+import { Environment } from '../../../shared/environment.js';
+import { createSpecialPageMessaging } from '../../../shared/create-special-page-messaging.js';
+import { init } from '../app/index.js';
```
Comment (mgurgel) [inline special-pages/pages/example/src/index.js:10]:
```suggestion
import { init } from '../app/index.js';
import '../../../shared/live-reload.js';
```

File: special-pages/pages/special-error/src/index.js [hunk 2]
```diff
@@ -5,10 +5,10 @@
  */
 
 import { createTypedMessages } from '@duckduckgo/messaging';
-import { Environment } from '../../../../shared/environment.js';
-import { createSpecialPageMessaging } from '../../../../shared/create-special-page-messaging.js';
+import { Environment } from '../../../shared/environment.js';
+import { createSpecialPageMessaging } from '../../../shared/create-special-page-messaging.js';
 import { mockTransport } from './mock-transport.js';
-import { init } from '../../app/index.js';
+import { init } from '../app/index.js';
```
Comment (mgurgel) [inline special-pages/pages/special-error/src/index.js:11]:
```suggestion
import { init } from '../app/index.js';
import '../../../shared/live-reload.js';
```

**Review by mgurgel — DISMISSED** [review]
This is brilliant 🤩 . Only missing live-reload setup for example and special-error, but looks good otherwise. Works really well locally!

**Pull request merged**